openraft/src/engine/command.rs

```diff
@@ -219,30 +220,60 @@ where NID: NodeId
 
 /// A command to send return value to the caller via a `oneshot::Sender`.
 #[derive(Debug)]
-#[derive(PartialEq, Eq)]
 #[derive(derive_more::From)]
-pub(crate) enum Respond<NID, N>
+pub(crate) enum Respond<R, NID, N>
 where
     NID: NodeId,
     N: Node,
+    R: AsyncRuntime,
+{
+    Vote(ValueSender<R, Result<VoteResponse<NID>, Infallible>>),
+    AppendEntries(ValueSender<R, Result<AppendEntriesResponse<NID>, Infallible>>),
+    ReceiveSnapshotChunk(ValueSender<R, Result<(), InstallSnapshotError>>),
+    InstallSnapshot(ValueSender<R, Result<InstallSnapshotResponse<NID>, InstallSnapshotError>>),
+    InstallCompleteSnapshot(ValueSender<R, Result<SnapshotResponse<NID>, Infallible>>),
+    Initialize(ValueSender<R, Result<(), InitializeError<NID, N>>>),
+}
+
+impl<R, NID, N> PartialEq for Respond<R, NID, N>
+where
+    NID: NodeId + PartialEq,
+    N: Node + PartialEq,
+    R: AsyncRuntime,
```

Comment:
If we use the `derive(PartialEq)` it forces a constraint `PartialEq` on every generic, that why I expanded the macro. (same for Eq)